Sources/MarathonCore/Create.swift

```diff
@@ -38,25 +38,74 @@ extension CreateError: PrintableError {
 
 internal final class CreateTask: Task, Executable {
     private typealias Error = CreateError
-
+    
+    private var shouldOpen: Bool {
+        return !argumentsContainNoOpenFlag
+    }
+    
     func execute() throws {
         guard let path = arguments.first?.asScriptPath() else {
             throw Error.missingName
         }
-
+        
+        if arguments.contains("--tests") {
+            if let scriptFile = try? FileSystem().currentFolder.file(atPath: path),
+                let script = try? scriptManager.script(at: scriptFile.path) {
+                try createScriptTests(for: script)
+                try beginEditing(script, if: shouldOpen)
+            } else {
+                let file = try createScript(at: path)
+                let script = try scriptManager.script(at: file.path)
+                try createScriptTests(for: script)
+                try beginEditing(script, if: shouldOpen)
+            }
+        } else {
+            let file = try createScript(at: path)
```

Comment:
There are three combinations of actions we can take: 
1. Create a new script without tests
2. Create a new script and tests together
3. Create tests for an existing script (without overwriting the script)

Hence the three branches. Some of the surrounding logic can definitely be simplified, though.